src/frontend/src/components/common/Header/Header.tsx

```diff
@@ -24,27 +25,36 @@ function Header({ user }: Props) {
     localStorage.getItem('access_token'),
   );
   const [showModal, setShowModal] = useState(false);
-  const [userInfo, setUserInfo] = useState({ name: '' });
   const [message, setMessage] = useState('');
   const code = searchParams.get('code');
-
-  const { keycloakToken, error: authCodeError } = useFetchAuthCode(code);
+  const didAuthRef = useRef(false);
   const { keycloakUserDetail, error: userInfoError } =
     useFetchKeycloakUserInfo(accessToken);
 
   // https://github.com/microsoft/TypeScript/issues/48949
   // workaround
   const win: Window = window;
   useEffect(() => {
-    if (keycloakToken !== undefined && keycloakToken !== null) {
-      storeTokens(keycloakToken);
-      setAuthenticated(true);
-
-      setAccessToken(keycloakToken.access_token);
-      navigate(routes.PPQ_LANDING_PAGE);
-    } else return;
+    async function startFetching() {
+      if (didAuthRef.current === false) {
+        try {
+          didAuthRef.current = true;
+          const res = await getAccessToken(code);
+          const keycloakToken = await res?.json();
+          if (keycloakToken !== undefined) {
+            storeTokens(keycloakToken);
+            setAuthenticated(true);
+            setAccessToken(keycloakToken.access_token);
+            navigate(routes.PPQ_LANDING_PAGE);
+          }
+        } catch (e) {
+          console.log(e);
+        }
+      }
+    }
+    startFetching();
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can we can get rid of these eslint-disable comments now that our dependencies have shrunk.